vali_objects/utils/mdd_checker.py

```diff
@@ -185,14 +188,68 @@ def _get_sources_for_order(order, trade_pair, is_last_order):
             return position
 
 
+    def add_manual_flat_orders(self, hotkey:str, sorted_positions:List[Position], candle_data:Dict[TradePair, List[PriceSource]]) -> bool:
+        """
+        Add flat orders to the positions for a miner that has been eliminated.
+        """
+        success = True
+        any_changes_attempted = False
+        corresponding_elimination = [x for x in self.eliminations if hotkey == x['hotkey']][0]
```

Comment:
Should we check that [x for x in self.eliminations if hotkey == x['hotkey']] is nonempty before indexing into it?

Reply:
Yes we do but it is happening before the function. To be more clear, I refactored the function to pass in the corresponding elimination instead of recalculating it. Good catch.